udata_recommendations/tasks.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import json
+import importlib
+
+import mongoengine
+import requests
+import jsonschema
+from flask import current_app
+
+from udata.core.dataset.models import Dataset
+
+from udata.tasks import job
+from udata.commands import success, error
+
+log = logging.getLogger(__name__)
+
+
+def recommendations_clean():
+    nb_datasets = Dataset.objects.filter(**{
+        f'extras__recommendations:sources__exists': True,
+    }).update(**{
+        'unset__extras__recommendations': True,
+        'unset__extras__recommendations:sources': True,
+    })
+    success(f"Removed recommendations from {nb_datasets} dataset(s)")
+
+
+def recommendations_clean_source(source):
```

Comment:
Why keep the source-based logic? I think `recommendations_clean` is enough, just trigger it at the beginning of `run_recommendations_add`. With the added bonus of deleting the recommendations linked to a source that has been deleted.

Reply:
If you don't believe that it's useful, I'm deleting it :) Less code is good.

Reply:
🙏